linetools/spectralline.py

```diff
@@ -0,0 +1,370 @@
+            except AttributeError:
+                pass
+            else:
+                sig = sig / self.analy['spec'].conti[pix]
+
+        # Return
+        return fx, sig, wave
+
+
+    # EW 
+    def box_ew(self):
+        """  EW calculation
+        Default is simple boxcar integration
+        Observer frame, not rest-frame
+          WVMNX must be set!
+          spec must be set!
+
+        Parameters
+        ----------
+
+        Returns:
+          EW, sigEW : EW and error in observer frame
+        """
+        # Cut spectrum
+        fx, sig, wv = self.cut_spec(normalize=True)
+
+        # dwv
+        dwv = wv - np.roll(wv,1)
+        dwv[0] = dwv[1]
+
+
+        # Simple boxcar
+        EW = np.sum( dwv * (1. - fx) ) 
+        varEW = np.sum( dwv**2 * sig**2 )
+        sigEW = np.sqrt(varEW) 
+
+
+        # Fill
+        self.attrib['EW'] = EW 
+        self.attrib['sigEW'] = sigEW 
+
+        # Return
+        return EW, sigEW
+            
+    # EW 
+    def restew(self):
+        """  Rest EW calculation
+        Return rest-frame.  See "box_ew" above for details
+        """
+        # Standard call
+        EW,sigEW = self.box_ew()
+        # Push to rest-frame
+        self.attrib['EW'] = EW / (self.attrib['z']+1)
+        self.attrib['sigEW'] = sigEW / (self.attrib['z']+1)
+
+        # Return
+        return self.attrib['EW'], self.attrib['sigEW'] 
+
+    # Output
+    def __repr__(self):
+        txt = '[{:s}:'.format(self.__class__.__name__)
+        try:
+            txt = txt+' {:s},'.format(self.data['name'])
+        except KeyError:
+            pass
+        txt = txt + ' wrest={:g}'.format(self.wrest)
+        txt = txt + ']'
+        return (txt)
+
+# Class for Generic Absorption Line System
+class AbsLine(SpectralLine):
+    """Spectral absorption line
+    """
+    # Initialize with a .dat file
+    def __init__(self, wrest, linelist=None): 
+        # Generate with type
+        SpectralLine.__init__(self,'Abs', wrest, linelist=linelist)
+
+    def print_specline_type(self):
+        """"Return a string representing the type of vehicle this is."""
+        return 'AbsLine'
+
+    def fill_data(self,linelist=None):
+        ''' Fill atomic data and setup analy
+        Parameters:
+        -----------
+        linelist : LineList, optional
+          Class of linelist or str setting LineList
+        '''
+
+        # Deal with LineList
+        if linelist is None:
+            self.llist = LineList('ISM')
+        elif isinstance(linelist,basestring):
+            self.llist = LineList(linelist)
+        elif isinstance(linelist,LineList):
+            self.llist = linelist
+        else:
+            raise ValueError('Bad input for linelist')
+
+        # Data
+        self.data.update(self.llist[self.wrest])
+
+        #
+        self.analy.update( {
+            'FLG_EYE': 0,
+            'FLG_LIMIT': 0, # No limit
+            'DATFIL': '', 
+            'IONNM': self.data['name']
+            })
```

Comment:
Please lower case all of FLG_EYE, FLG_LIMIT, DATFIL, IONNM.  Also could we use 'datafile' instead of 'datfil'? And just 'name' or 'ionname' instead of 'ionnm'?